packages/react/src/components/HorizontalRule/HorizontalRule.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import classnames from 'classnames';
+import settings from 'carbon-components/es/globals/js/settings';
+
+import '@ibmdotcom/styles/scss/components/horizontalrule/_horizontalrule.scss';
+
+const { prefix } = settings;
+/**
+ * Horizontal Rule component
+ *
+ * @param {object} props props object
+ * @param {string} props.type type of rule (solid or dashed)
+ * @param {string} props.color color of the rule (low-contrast, medium-contrast, or high-contrast)
+ * @param {string} props.width width of the rule (thin or thick)
+ * @param {string} props.length length of rule (small, medium, large, inset, or overhung)
```

Comment:
@photodow 
> `props.length -> props.size // props.fixed`

Is it possible to separate the fluid vs fixed rules? Especially because it looks like Wonil isn't using the dashed line in the fixed lines. We might need to confirm if fixed should allow for `props.weight` as well, in the specs it has only one weight.

Fluid: would have inset / overhung
Fixed: small, medium, large.

Reply:
@oliviaflory I've mentioned in other comments that I don't think fluid should be a prop since `<hr />` is fluid by default. Basically, if they want to fix/size it they can then override it.

```jsx
{/* fluid by default */}
<hr />
```

```jsx
{/* set fixed prop */}
<hr size="large" />
```

Reply:
@oliviaflory I removed `inset` and `overhung` from the `size` prop so now only have 

size: small, medium, large, fluid

only when we have both `dashed` and `fluid` classes do we apply the dash style 
```
.#{$prefix}--hr--dashed.#{$prefix}--hr--fluid
```

only when we have the `fluid` and `thick` style do we allow for the weight to increase
```
.#{$prefix}--hr--fluid.#{$prefix}--hr--thick {
        height: carbon--rem(2px);
 }
```